File: customermanagementbackend/src/main/java/com/customermanagementbackend/service/CustomerService.java
package com.customermanagementbackend.service;

import com.customermanagementbackend.model.Customer;
import com.customermanagementbackend.repository.CustomerRepository;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.data.domain.Sort;
import org.springframework.stereotype.Service;

import java.util.List;
import java.util.Optional;

@Service
public class CustomerService implements CustomerServiceInterface {

    @Autowired
    private CustomerRepository customerRepository;

    @Override
    public Customer saveCustomer(Customer customer) {
        return customerRepository.save(customer);
    }

    @Override
    public Optional<Customer> getCustomerById(int id) {
        return customerRepository.findById(id);
    }

    @Override
    public List<Customer> getAllCustomers() {
        return customerRepository.findAll(Sort.by(Sort.Direction.DESC, "id"));
    }

    @Override
    public Customer updateCustomer(int id, Customer customer) {
        Customer customerToUpdate = customerRepository.findById(id).orElseThrow();
        customerToUpdate.setFirstName(customer.getFirstName());
        customerToUpdate.setLastName(customer.getLastName());
        customerToUpdate.setEmail(customer.getEmail());
        return customerRepository.save(customerToUpdate);
    }

    @Override
    public void deleteCustomer(int id) {
        customerRepository.deleteById(id);
    }
}

File: customermanagement-frontend/src/service/CustomerService.js
import axios from "axios";

const BASE_URL = "http://localhost:8080/customer";
class CustomerService{

    /**Method to get all customers from our api or database */
    getAllCustomers(){
        return axios.get(BASE_URL);
    }
    /**Method to save customer */
    saveCustomer(customerData){
        return axios.post(BASE_URL, customerData);
    }
    updateCustomer(id, customerData){
        return axios.put(`${BASE_URL}/${id}`, customerData)
    }
    getCustomerById(id){
        return axios.get(`${BASE_URL}/${id}`);
    }
    deleteCustomer(id){
        return axios.delete(BASE_URL +"/" +id);
    }

}
export default new CustomerService();